src/playground/app.js

```diff
@@ -538,15 +602,211 @@ function updateStageButtons() {
+
+  // Remove existing diagnostic summary
+  const existingSummary = editorHeader.querySelector(".diagnostic-summary");
+  if (existingSummary) {
+    existingSummary.remove();
+  }
+
+  if (!lastDiagnostics) {
+    return;
+  }
+
+  let totalErrors = 0;
+  let totalWarnings = 0;
+
+  // Check if we have the new format with summary
+  if (lastDiagnostics.summary) {
+    totalErrors = lastDiagnostics.summary.errors;
+    totalWarnings = lastDiagnostics.summary.warnings;
+  } else {
+    // Fallback to old format counting
+    Object.values(lastDiagnostics).forEach((stageDiagnostics) => {
+      if (Array.isArray(stageDiagnostics)) {
+        stageDiagnostics.forEach((diagnostic) => {
+          if (
+            diagnostic.severity === "error" ||
+            diagnostic.severity === "fatal"
+          ) {
+            totalErrors++;
+          } else if (diagnostic.severity === "warning") {
+            totalWarnings++;
+          }
+        });
+      }
+    });
+  }
+
+  // Always show summary after compilation (when timing info is available)
+  if (lastCompileTime !== null) {
+    const summaryDiv = document.createElement("div");
+    summaryDiv.className = "diagnostic-summary";
+
+    let summaryText = "";
+    // Always show error/warning count after compilation
+    summaryText += `Found ${totalErrors} error(s) and ${totalWarnings} warning(s)`;
+
+    if (lastCompileTime !== null) {
+      let timeText;
+      if (lastCompileTime < 1000) {
+        timeText = `${Math.round(lastCompileTime)}ms`;
+      } else {
+        timeText = `${(lastCompileTime / 1000).toFixed(1)}s`;
+      }
+      summaryText += (summaryText ? " • " : "") + `⚡ ${timeText}`;
+    }
+
+    summaryDiv.innerHTML = summaryText;
+    editorHeader.appendChild(summaryDiv);
+  }
+}
+
+// Auto-compile setup
+let compileTimeout;
+let compileStartTime = null;
+
+function setupAutoCompile() {
+  const editor = document.getElementById("editor");
+  if (editor) {
+    editor.addEventListener("input", () => {
+      // Debounce compilation to avoid excessive calls
+      clearTimeout(compileTimeout);
+      compileStartTime = performance.now(); // Start timing when user stops typing
+      compileTimeout = setTimeout(() => {
+        if (currentState === "READY" || currentState === "LOADED") {
+          compileCode();
+        }
+      }, 20); // 20ms delay for better responsiveness
+    });
+  }
+}
+
+// URL sharing functionality
+function hashContent(content) {
+  if (!content || content.length > 10000) return null; // Don't hash very large content
+
+  let hash = 0;
+  for (let i = 0; i < content.length; i++) {
+    const char = content.charCodeAt(i);
+    hash = (hash << 5) - hash + char;
+    hash = hash & hash; // Convert to 32-bit integer
+  }
+  return Math.abs(hash).toString(36);
+}
+
+function updateUrl(content) {
+  const hash = hashContent(content);
+  if (hash) {
+    const newUrl = `${window.location.origin}${window.location.pathname}?content=${hash}`;
```

Comment:
you likely want `#content` instead of query parameter `?content` because the first option doesn't reach server, living only on the client side